sympy/combinatorics/fp_groups.py

```diff
@@ -1209,11 +1232,14 @@ def elimination_technique_2(C):
     C._schreier_generators = gens
     return C._schreier_generators, C._reidemeister_relators
 
-def reidemeister_presentation(fp_grp, H, C=None):
+def reidemeister_presentation(fp_grp, H, C=None, homomorphism=False, subgroup=False):
     """
     fp_group: A finitely presented group, an instance of FpGroup
     H: A subgroup whose presentation is to be found, given as a list
     of words in generators of `fp_grp`
+    homomorphism: When set to True, return a homomorphism from the subgroup
+                    to the parent group
+    subgroup: Returns the subgroup defined on the presentation
```

Comment:
That's what `subgroup()` is for. As I've said in the other PR, I'd be inclined to just return `C.schreier_gen_elem` here when `homomorphism` is true and build the actual homomorphism in `subgroup`. If somebody wants to have a homomorphism, they will almost certainly want a subgroup as well - otherwise there's nothing to apply the homomorphism to.